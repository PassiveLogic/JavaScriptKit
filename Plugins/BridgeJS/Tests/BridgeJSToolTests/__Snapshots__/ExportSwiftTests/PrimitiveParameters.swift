// NOTICE: This is auto-generated code by BridgeJS from JavaScriptKit,
// DO NOT EDIT.
//
// To update this file, just rebuild your project or run
// `swift package bridge-js`.

@_spi(JSObject_id) import JavaScriptKit

#if arch(wasm32)
@_extern(wasm, module: "bjs", name: "return_string")
private func _return_string(_ ptr: UnsafePointer<UInt8>?, _ len: Int32)
@_extern(wasm, module: "bjs", name: "init_memory")
private func _init_memory(_ sourceId: Int32, _ ptr: UnsafeMutablePointer<UInt8>?)

@_extern(wasm, module: "bjs", name: "swift_js_retain")
private func _swift_js_retain(_ ptr: Int32) -> Int32
@_extern(wasm, module: "bjs", name: "swift_js_throw")
private func _swift_js_throw(_ id: Int32)
#endif

@_expose(wasm, "bjs_check")
@_cdecl("bjs_check")
public func _bjs_check(a: Int32, b: Float32, c: Float64, d: Int32) -> Void {
    #if arch(wasm32)
    check(a: Int(a), b: b, c: c, d: d == 1)
    #else
    fatalError("Only available on WebAssembly")
    #endif
}