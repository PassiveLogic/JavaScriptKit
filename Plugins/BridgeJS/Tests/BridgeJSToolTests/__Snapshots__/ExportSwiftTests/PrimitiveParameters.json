{
  "classes" : [

  ],
  "functions" : [
    {
      "abiName" : "bjs_check",
      "name" : "check",
      "parameters" : [
        {
          "label" : "a",
          "name" : "a",
          "type" : {
            "int" : {

            }
          }
        },
        {
          "label" : "b",
          "name" : "b",
          "type" : {
            "float" : {

            }
          }
        },
        {
          "label" : "c",
          "name" : "c",
          "type" : {
            "double" : {

            }
          }
        },
        {
          "label" : "d",
          "name" : "d",
          "type" : {
            "bool" : {

            }
          }
        }
      ],
      "returnType" : {
        "void" : {

        }
      }
    }
  ]
}